code/modules/client/verbs/ooc.dm

```diff
@@ -272,43 +274,123 @@ GLOBAL_VAR_INIT(normal_ooc_colour, "#002eb8")
+			// Show us the player's mob name in the list in front of their displayed key
+			// Add the player's displayed key to the list
+			players["[C.mob]([displayed_key])"] = displayed_key
+
+		// Add the player's displayed key to the list if we or the player aren't a ghost or they're using a fakekey
 		else
-			choices[displayed_choicename] = C
-	choices = sortList(choices)
-	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in choices
-	if(!selection || !(selection in choices))
+			players[displayed_key] = displayed_key
+
+	// Check if the list is empty
+	if(!players.len)
+		// Express that there are no players we can ignore in chat
+		to_chat(src, "There are no other players you can ignore!")
+
+		// Stop running
+		return
+
+	// Sort the list
+	players = sortList(players)
+
+	// Request the player to ignore
+	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in players
+
+	// Stop running if we didn't receieve a valid selection
+	if(!selection || !(selection in players))
+		return
+
+	// Store the selected player
+	selection = players[selection]
+
+	// Check if the selected player is not on our ignore list
+	if(!(selection in prefs.ignoring))
+		// Add the selected player to our ignore list
+		prefs.ignoring += selection
+
+		// Save our preferences
+		prefs.save_preferences()
+
+	// Express that we've ignored the selected player in chat
+	to_chat(src, "You are now ignoring [selection] on the OOC channel.")
+
+// Unignore verb
+/client/verb/select_unignore()
+	set name = "Unignore"
+	set category = "OOC"
+	set desc = "Stop ignoring a player's messages on the OOC channel"
+
+	// Check if we've ignored any players
+	if(!prefs.ignoring.len)
+		// Express that we haven't ignored any players in chat
+		to_chat(src, "You haven't ignored any players!")
+
+		// Stop running
 		return
-	displayed_choicename = selection // ckey string
-	selection = choices[selection] // client
-	if(selection == src)
-		to_chat(src, "You can't ignore yourself.")
+
+	// Make a list for players we've ignored
+	var/list/players = list()
+
+	// Add every player we've ignored to the list
+	for(var/player in prefs.ignoring)
+		players += player
+
+	// Sort the list
+	players = sortKey(players)
+
+	// Request the player to unignore
+	var/selection = input("Please, select a player!", "Unignore", null, null) as null|anything in players
+
+	// Stop running if we didn't receive a valid selection
+	if(!selection || !(selection in players))
 		return
-	ignore_key(selection, displayed_choicename)
+
+	// Check if the selected player is on our ignore list
+	if(selection in prefs.ignoring)
+		// Remove the selected player from our ignore list
+		prefs.ignoring -= selection
+
+		// Save our preferences
+		prefs.save_preferences()
+
+	// Express that we've unignored the selected player in chat
+	to_chat(src, "You are no longer ignoring [selection] on the OOC channel.")
```

Comment:
Would be nice if you only got positive feedback if it actually removed someone from ignoring.